hadoop-ozone/tools/src/main/java/org/apache/hadoop/ozone/repair/om/OMRatisLogRepair.java

```
@@ -0,0 +1,190 @@
+import java.lang.reflect.InvocationTargetException;
+import java.nio.ByteBuffer;
+import org.apache.hadoop.hdds.cli.HddsVersionProvider;
+import org.apache.hadoop.ozone.om.helpers.OMRatisHelper;
+import org.apache.hadoop.ozone.protocol.proto.OzoneManagerProtocolProtos;
+import org.apache.hadoop.ozone.repair.RepairTool;
+import org.apache.ratis.proto.RaftProtos;
+import org.apache.ratis.server.metrics.SegmentedRaftLogMetrics;
+import org.apache.ratis.server.raftlog.segmented.LogSegmentPath;
+import org.apache.ratis.server.raftlog.segmented.LogSegmentStartEnd;
+import org.apache.ratis.server.raftlog.segmented.SegmentedRaftLogInputStream;
+import org.apache.ratis.server.raftlog.segmented.SegmentedRaftLogOutputStream;
+import org.apache.ratis.util.Preconditions;
+import org.apache.ratis.util.SizeInBytes;
+import picocli.CommandLine;
+
+
+/**
+ * Tool to omit a raft log in a ratis segment file.
+ */
+@CommandLine.Command(
+    name = "ratislogrepair",
+    description = "CLI to omit a raft log in a ratis segment file. " +
+        "Requires admin privileges.",
+    mixinStandardHelpOptions = true,
+    versionProvider = HddsVersionProvider.class
+)
+public class OMRatisLogRepair extends RepairTool {
+
+  @CommandLine.Option(names = {"-s", "--segment-path", "--segmentPath"},
+      required = true,
+      description = "Path of the input segment file")
+  private File segmentFile;
+  @CommandLine.Option(names = {"-o", "--output-path", "--outputPath"},
+      required = true,
+      description = "Path of the repaired segment file")
+  private String outputPath;
+
+  @CommandLine.Option(names = {"--index"},
+      required = true,
+      description = "Path of the segment file")
+  private long index;
+
+  @Override
+  public void execute() throws Exception {
+
+    LogSegmentPath pi = LogSegmentPath.matchLogSegment(this.segmentFile.toPath());
+    if (pi == null) {
+      error("Invalid segment file");
+      throw new RuntimeException("Invalid Segment File");
+    }
```

Can we also fail early if outputPath is null?

---

In the createOutputFile(), if there is some issue with the output file it fails. It happens before we start processing, so it fails early.